### DISCUSSION TITLE
Add PyMuPDF to server

### DISCUSSION THREAD

**Discussion by jonsken** [pymupdf/PyMuPDF#1637] [Looking for help] [answered]
I want to use PyMuPDF in a project and therefore I need to add the library (+fitz) to a webserver. But unfortunately I do not have permission to install anything on this webserver... 
Is there a way to link or integrate it without `pip install` ? Maybe add it as a directory inside my project and then link it, would this be possible?

**Comment by JorjMcKie** [comment, marked as answer]
I am probably not the most adequate person to answer this.
So please everyone, take this as an invitation!

@jonsken - obviously PyMuPDF must run und control of a Python interpreter and must be installed in a Python environment. So your question is more like "How can your webserver communicate with a remote Python environment?"
Not a PyMuPDF question.
You would probably have to start a separate (local?) webserver running a script that listens and accepts requests for PyMuPDF calls, and then responds accordingly.
Just immediate thoughts ...

**Reply by jonsken** [reply]
new approach: Is PyMuPDF "online accessible" on some kind of server? So I kind of just pass in the arguments and the script executes somewhere else.
Just asking before I start to set up something by myself

**Reply by JorjMcKie** [reply]
That was more or less what I was pointing to - but lots of details must be worked out of course.
Why don't you start with two local web servers on your machine and develop a suitable communication protocol between them.
Maybe you can even use Python's `multiprocessing` module, quote:

![grafik](https://user-images.githubusercontent.com/8290722/157863823-00db87b2-b1bb-457d-a0b4-b2719b030375.png)

I must stress again: by no means I am an expert in this.
